File: LeetCode/剑指Offer/04.py
class Solution:
    def findNumberIn2DArray(self, matrix: List[List[int]], target: int) -> bool:
        if len(matrix)==0 or len(matrix[0])==0:
            return False
        n=0
        m=len(matrix[0])-1
        while n<len(matrix) and m>=0:
            if matrix[n][m]==target:
                return True
            elif matrix[n][m]>target:
                m-=1
            else:
                n+=1
        return False

File: LeetCode/剑指Offer/04.js
/**
 * @param {number[][]} matrix
 * @param {number} target
 * @return {boolean}
 */
var findNumberIn2DArray = function(matrix, target) {
    if(matrix.length===0 || matrix[0].length===0){return false;}
    var n=0;
    var m=matrix[0].length-1;
    while(n<matrix.length && m>=0){
            if(matrix[n][m]===target){return true;}
            else if(matrix[n][m]>target){m--;}
            else{n++;}
        }
        return false;
};
